### PR TITLE
[IN-329][Reviews] Add withdrawal view to moderation list and detail pages

### PR DESCRIPTION
**Pull request by baylee-d**
## Purpose
To add withdrawal views and actions to the `moderation-list` and the `preprint-detail` pages.

## Summary of Changes/Side Effects
- Add `Withdrawn` section to the `Submissions` tab in the `moderation-list` page
- Update `preprint-detail` page to be a tombstone page if the preprint is withdrawn
- Update `preprint-status-banner` to have the withdrawn reason instead of the action dropdown if the preprint is withdrawn (or show `no reason` if a reason wasn't provided)
- Add the ability to withdraw a preprint on post-moderation services

### Moderation List
There should be a new tab under `Submissions` for withdrawn preprints.  It should include all withdrawn preprints with a circle-dash icon.  The wording should be `Withdrawn` or `Withdrawn on`, whichever wording makes sense for the datetime.  It should also have the moderator withdrew it.

### Preprint-detail
If the preprint has not been withdrawn and the service is a post-moderation service, then you should be able to have the option to accept or withdraw the preprint instead of accept or remove.  It should give you the option to add a review comment.  When you withdraw the preprint, it should take you back to the moderation list and you should see your preprint under the `Withdrawals` tab.

If the preprint is withdrawn you should be seeing a tombstone page.  The banner should have a withdrawn status icon, the withdrawal date and who withdrew it, and a `View reason` button.  If there isn't a reason provided with the withdrawal, the button will be disabled and a popover will show on hover that says "No reason provided."

## Screenshots

On post-moderation services, you'll have the option to withdraw a preprint:
screen shot 2018-08-06 at 10 09 49 am

Withdrawn preprints will show up in the `moderation-list` under the `Withdrawn` tab:
screen shot 2018-08-06 at 1 52 17 pm

Withdrawn preprints should now have a tombstone page with an updated status banner:
screen shot 2018-08-06 at 10 11 22 am

If there's a reason provided it'll be shown in the dropdown:
screen shot 2018-08-06 at 10 11 09 am

Otherwise the button will be disabled and there will be a small popover saying none was provided:
screen shot 2018-08-06 at 10 18 42 am

## Ticket

https://openscience.atlassian.net/browse/IN-329

## Notes for Reviewer
N/A

# Reviewer Checklist

- [ ] meets requirements
- [ ] easy to understand
- [ ] DRY
- [ ] testable and includes test(s)
- [ ] changes described in `CHANGELOG.md`

### CODE REVIEW

File: app/components/preprint-status-banner/component.js [hunk 1]
```diff
@@ -124,6 +132,30 @@ export default Component.extend({
...
     userActivity: computed.or('commentEdited', 'decisionValueToggled'),
 
+    labelDate: computed('submission', function() {
+        return this.get('submission.dateWithdrawn') ?
+            this.get('submission.dateWithdrawn') :
+            this.get('submission.dateLastTransitioned');
+    }),
+
+    labelDecision: computed('submission', function() {
+        return this.get('submission.isPublished') === true ?
```
Comment (alexschiller) [inline app/components/preprint-status-banner/component.js]: if this is always a bool, it probably doesn't need the `=== true` part

File: app/components/preprint-status-banner/component.js [hunk 2]
```diff
@@ -124,6 +132,30 @@ export default Component.extend({
...
+            'components.preprintStatusBanner.decision.withdrawn.label' :
+            'components.preprintStatusBanner.decision.reject.label';
+    }),
+
+    radioDecision: computed('submission', function() {
+        return this.get('submission.isPublished') === true ?
```
Comment (alexschiller) [inline app/components/preprint-status-banner/component.js]: same, `=== true` is probably not needed.

File: app/templates/preprints/provider/preprint-detail.hbs [hunk 3]
```diff
@@ -26,143 +26,212 @@
...
+                                    <a type="button" class="btn btn-info" href={{node.links.html}}>
+                                        {{t "content.projectButton.button"}}
+                                    </a>
+                                </div>
+                            </div>
                         {{/if}}
+                    </div>
+                </div>
+            {{else}}
+                <div class="row p-md" style="margin-bottom: 50px">
```
Comment (alexschiller) [inline app/templates/preprints/provider/preprint-detail.hbs]: inline style? should this be on stylesheet?

**Comment by alexschiller** [comment]
Did a quick pass over, Yuhuai is going to do a pass and I'll look it over one more time before merge.

**Review by adlius — COMMENTED** [review]
Checked locally, everything works fine when the NoneIfWithdrawal for `reviews_state` is removed. Also, `documentType` needs to be pased to `content.preprintDoi` translation string. Other than that, LGTM.

**Comment by adlius** [comment]
Closing this as this is cherry-picked over to #152.

**Pull request closed without merging**